extensions/dicom-microscopy/src/utils/cleanDenaturalizedDataset.ts

```diff
@@ -38,7 +44,15 @@ export default function cleanDenaturalizedDataset(obj: any): any {
         delete obj[key].Value;
       } else if (Array.isArray(obj[key].Value) && obj[key].vr) {
         if (obj[key].Value.length === 1 && obj[key].Value[0].BulkDataURI) {
-          obj[key].BulkDataURI = obj[key].Value[0].BulkDataURI;
+          // If the BulkDataURI is not a study level URI, then we need to convert the
+          // relative URI to an absolute URI (see open PR https://github.com/dcmjs-org/dicomweb-client/pull/54)
+          if (!obj[key].Value[0].BulkDataURI.includes('studies/')) {
```

Comment:
You need to call fixBulkDataURI here - the test here as you have it isn't safe, because a perfectly value bulkdata URI is:
https://server/bulkdata/hashcode.dat
There is NO requirement at all that the bulkdata URI contain the studies path.  IF it is a path relative URI, then it will be relative to the metadata call that was used to retrieve it, but nothing says it has to be in the dicomweb path even.